src/bsp/src/stm32/hal/decawave/platform/stm32f429zi/src/deca_spi.c

```diff
@@ -0,0 +1,68 @@
+#include "deca_device_api.h"
+#include "hivemind_hal.h"
+#include "main.h"
+#include "stm32f4xx_hal_def.h"
+
+/**
+ * @brief Low-level SPI function used by Decawave driver to write bytes to the device
+ * @param headerLength Length of header in bytes
+ * @param headerBuffer Buffer of the header to send
+ * @param bodyLength Length of body in bytes
+ * @param bodyBuffer Buffer of the body to send
+ * @return unused (only here for compatibility with deca driver)
+ */
+int writetospi(uint16_t headerLength,
+               const uint8_t* headerBuffer,
+               uint32_t bodyLength,
+               const uint8_t* bodyBuffer) {
+    decaIrqStatus_t stat;
+    stat = decamutexon();
+
+    while (HAL_SPI_GetState(DW_SPI) != HAL_SPI_STATE_READY) {
+    }
+
+    HAL_GPIO_WritePin(DW_NSS_A_GPIO_Port, DW_NSS_A_Pin, GPIO_PIN_RESET);
+
+    HAL_SPI_Transmit(DW_SPI, (uint8_t*)&headerBuffer[0], headerLength, HAL_MAX_DELAY);
+    HAL_SPI_Transmit(DW_SPI, (uint8_t*)&bodyBuffer[0], bodyLength, HAL_MAX_DELAY);
```

Comment:
This is definitely out of scope for this PR, but will we use DMA or interrupts eventually? Or eventually periodically get distances of neighbours via an interrupt or a software timer?

Reply:
We will definitely use interrupts and timers, but it should not affect the SPI itself. The DW has an interrupt line to notify when an RX event happens instead of polling status registers, but the underlying SPI will remain blocking as we only send very small payloads over it and the deca driver was made to have blocking SPI calls.